### PR TITLE
Automated testing setup

### PR DESCRIPTION
**Pull request by serge1peshcoff**
Well I finally implemented some kind of automated testing.
What it is supposed to do:
<s>
- clone the `oms-docker`, `oms-core` and `oms-serviceregistry`
- starts all of these services
- waits for the setup to bootstrap
- runs all the tests and sends a Coveralls report
- stops the OMS setup.

Most likely not usable for testing on local PC (as it starts another OMS setup, so there will probably be some ports conflicts, also it needs MongoDB started), so it's only Travis CI-specific (I'll probably think about how to set it up for local testing later).
</s>
I've rewrote it, so now it stubs `omscore` and `omsserviceregistry` and does requests to these stubs instead of real applications.

<s>This PR is not for merging yet (apparently it's already broken, since I've got the Slack notification for broken build), I've created it for testing Travis CI builds.</s> The tests are now passing, only thing left to do is increase tests coverage and integrate all of this with frontend.

### CODE REVIEW

**Comment by i3anaan** [comment]
Cool to see some tests!

I am somewhat confused where the actual tests are, however (you set tests_ok to true in `test/setup.sh` ?)
Also, how come you do not seem to call any tests to run in travis.yml? (Mostly making sure here you are not running tests on _every_ install of oms-events.

Finally, as you said it is not working yet, you aware of that so no further comment needed here I think :)

**Comment by serge1peshcoff** [comment]
@i3anaan Well, the tests themselves are in `test/api`, and this variable is needed to exit the script with error if the tests failed (we still need to send the coverage report after executing tests, so just using `./node_modules/.bin/nyc ./node_modules/mocha/bin/_mocha --timeout 10000 test/api/*.js || exit 1` won't work.
The `.travis.yml` executes `npm test` by default, and this task just runs `./test.setup.sh`, so it is executed on Travis.

**Comment by i3anaan** [comment]
@serge1peshcoff Answers my questions, thanks! All that is left is now to get it running correctly ;)

File: .coveralls.yml [hunk 1]
```diff
@@ -1 +0,0 @@
-repo_token: qpyDcfzamRoA7NkghlM1S8yLUk8H7Xd8U
```
Comment (linuxbandit) [inline .coveralls.yml]: Did you generate another one, if it needs to be secret? :D

Comment (serge1peshcoff) [reply .coveralls.yml]: @linuxbandit well I've switched from Coveralls to Codecov.io (that's why I don't need this one) and Codecov.io doesn't require repository token if it is running inside Travis

**Comment by i3anaan** [comment]
Wow the report on Travis looks really nice! Well done!

**Pull request merged**